tests/core/initializers_test.py

```diff
@@ -264,3 +265,256 @@ def test_gauss_initializer(self, lse_mode, rng: jnp.ndarray):
+      geom_y = geometry.Geometry(geom_y.cost_matrix, epsilon=eps)
+    else:
+      raise NotImplementedError(kind)
+    prob = quad_problems.QuadraticProblem(geom_x, geom_y)
+
+    solver = gromov_wasserstein.GromovWasserstein(
+        rank=rank, quad_initializer=None, kwargs_init=kwargs_init
+    )
+    initializer = solver.create_initializer(prob)
+
+    assert isinstance(initializer, quad_initializers.LRQuadraticInitializer)
+    assert initializer.rank == rank
+    linear_init = initializer._linear_lr_initializer
+    if kind in ("pc", "lrc"):
+      assert isinstance(linear_init, initializers_lr.KMeansInitializer)
+    else:
+      assert isinstance(linear_init, initializers_lr.RandomInitializer)
+    assert linear_init._kwargs == kwargs_init
+
+  def test_non_lr_initializer(self):
+    solver = gromov_wasserstein.GromovWasserstein(
+        rank=-1, quad_initializer="not used"
+    )
+    initializer = solver.create_initializer(prob="not used")
+    assert isinstance(initializer, quad_initializers.QuadraticInitializer)
+
+  @pytest.mark.parametrize("rank", [-1, 2])
+  def test_explicitly_passing_initializer(self, rank: int):
+    if rank == -1:
+      linear_init = init_lib.SortingInitializer()
+      quad_init = quad_initializers.QuadraticInitializer()
+    else:
+      linear_init = initializers_lr.Rank2Initializer(rank)
+      quad_init = quad_initializers.LRQuadraticInitializer(linear_init)
+
+    solver = gromov_wasserstein.GromovWasserstein(
+        initializer=linear_init,
+        quad_initializer=quad_init,
+    )
+
+    assert solver.linear_ot_solver.initializer is linear_init
+    assert solver.quad_initializer is quad_init
+    if solver.is_low_rank:
+      assert solver.quad_initializer.rank == rank
+
+  @pytest.mark.parametrize("eps", [0., 1e-2])
+  def test_gw_better_initialization_helps(self, rng: jnp.ndarray, eps: float):
+    n, m, d1, d2, rank = 123, 124, 12, 10, 5
+    key1, key2, key3, key4 = jax.random.split(rng, 4)
+
+    geom_x = pointcloud.PointCloud(
+        jax.random.normal(key1, (n, d1)),
+        jax.random.normal(key2, (n, d1)),
+        epsilon=eps,
+        scale_cost=1.,
+    )
+    geom_y = pointcloud.PointCloud(
+        jax.random.normal(key3, (m, d2)),
+        jax.random.normal(key4, (m, d2)),
+        epsilon=eps,
+        scale_cost=1.,
+    )
+    problem = quad_problems.QuadraticProblem(geom_x, geom_y)
+    solver_random = gromov_wasserstein.GromovWasserstein(
+        rank=rank,
+        initializer="random",
+        quad_initializer="random",
+        epsilon=eps,
+        store_inner_errors=True,
+    )
+    solver_kmeans = gromov_wasserstein.GromovWasserstein(
+        rank=rank,
+        initializer="k-means",
+        quad_initializer="k-means",
+        epsilon=eps,
+        store_inner_errors=True
+    )
+
+    out_random = solver_random(problem)
+    out_kmeans = solver_kmeans(problem)
+
+    assert out_random.reg_gw_cost - out_kmeans.reg_gw_cost >= 15.
```

Comment:
nice :)

Reply:
There was issue with the `KL` implementation, now it's just `1.0`